e2e/test/scenarios/actions/actions-on-dashboards.cy.spec.js

```diff
@@ -987,6 +992,55 @@ describe(
         });
       });
 
+      it("refetches form values when id changes (metabase#33084)", () => {
+        const actionName = "Update";
+
+        cy.get("@modelId").then(id => {
+          createImplicitAction({
+            kind: "update",
+            model_id: id,
+          });
+        });
+
+        createDashboardWithActionButton({
+          actionName,
+          idFilter: true,
+        });
+
+        filterWidget().click();
+        addWidgetStringFilter("5");
+
+        cy.findByRole("button", { name: actionName }).click();
+
+        cy.wait("@executePrefetch");
+
+        modal().within(() => {
+          cy.findByPlaceholderText("Team Name").should(
+            "have.value",
+            "Energetic Elephants",
+          );
+          cy.findByPlaceholderText("Score").should("have.value", "30");
+
+          cy.icon("close").click();
+        });
+
+        filterWidget().click();
+        popover().find("input").first().type("{backspace}10");
+        cy.button("Update filter").click();
+
+        cy.findByRole("button", { name: actionName }).click();
```

Comment:
This is the same as
```suggestion
        cy.button(actionName).click();
```

Reply:
35990d6777a6694eedb473abc58847de3249d2e7